pkg/controllers/testing/factories/keda_scaledobjectspec.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2020 the original author or authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package factories
+
+import (
+	"fmt"
+
+	kedav1alpha1 "github.com/projectriff/system/pkg/apis/thirdparty/keda/v1alpha1"
+)
+
+type kedaScaledObjectSpec struct {
+	target *kedav1alpha1.ScaledObjectSpec
+}
+
+func KedaScaledObjectSpec(seed ...*kedav1alpha1.ScaledObjectSpec) *kedaScaledObjectSpec {
+	var target *kedav1alpha1.ScaledObjectSpec
+	switch len(seed) {
+	case 0:
+		target = &kedav1alpha1.ScaledObjectSpec{}
+	case 1:
+		target = seed[0]
+	default:
+		panic(fmt.Errorf("expected exactly zero or one seed, got %v", seed))
+	}
+	return &kedaScaledObjectSpec{
+		target: target,
+	}
+}
+
+func (p *kedaScaledObjectSpec) deepCopy() *kedaScaledObjectSpec {
+	return KedaScaledObjectSpec(p.target.DeepCopy())
+}
+
+func (k *kedaScaledObjectSpec) Get() *kedav1alpha1.ScaledObjectSpec {
+	return k.deepCopy().target
+}
+
+func (k *kedaScaledObjectSpec) Mutate(m func(*kedav1alpha1.ScaledObjectSpec)) *kedaScaledObjectSpec {
+	k = k.deepCopy()
+	m(k.target)
+	return k
+}
+
+func (k *kedaScaledObjectSpec) ScaleTargetRefDeployment(deploymentName string) *kedaScaledObjectSpec {
```

Comment:
nit: for methods that take a single string that may be parametrized, I've been using the signature from `fmt.Sprintf(format string, a ...interface{})`.

https://github.com/projectriff/system/blob/27d19f98465a3d1c54f1c194ecae4c55d6e866dd/pkg/controllers/testing/factories/function.go#L158-L187

Reply:
Adopted, thanks.